src/include/graph/graph_mem.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "common/copy_constructors.h"
+#include "common/types/types.h"
+
+using namespace std;
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace graph {
+
+using weight_t = offset_t;
+constexpr weight_t DEFAULT_WEIGHT = 1.0;
+
+struct Neighbor {
+    offset_t neighbor;
+    weight_t weight;
+};
+
+// CSR-like in-memory representation of an undirected weighted graph. Insert nodes in sequence
+// by first calling `initNextNode()` and then insert all its neighbors using `insertNbr()`.
+// Undirected edges should be explicitly inserted twice.
+struct InMemGraph {
```

Comment:
I gave this feedback in the other comment but these need to be memory manager backed. Doing the 1.2x growth seems very simple, so I would do that.

Reply:
Addressed using a a custom allocator.